indexer/services/roundtable/src/helpers/aws.ts

```diff
@@ -33,13 +42,75 @@ export async function getMostRecentDBSnapshotIdentifier(rds: RDS): Promise<strin
     throw Error(`No DB snapshots found with identifier: ${config.RDS_INSTANCE_NAME}`);
   }
 
+  let snapshots: RDS.DBSnapshotList = awsResponse.DBSnapshots;
+  // Only include snapshots with snapshot identifier that starts with prefixInclude
+  if (snapshotIdentifierPrefixInclude !== undefined) {
+    snapshots = snapshots
+      .filter((snapshot) => snapshot.DBSnapshotIdentifier &&
+        snapshot.DBSnapshotIdentifier.startsWith(snapshotIdentifierPrefixInclude),
+      );
+  }
+  if (snapshotIdentifierPrefixExclude !== undefined) {
+    snapshots = snapshots
+      .filter((snapshot) => snapshot.DBSnapshotIdentifier &&
+        !snapshot.DBSnapshotIdentifier.startsWith(snapshotIdentifierPrefixExclude),
+      );
+  }
+
   logger.info({
     at: `${atStart}getMostRecentDBSnapshotIdentifier`,
     message: 'Described snapshots for database',
-    mostRecentSnapshot: awsResponse.DBSnapshots[awsResponse.DBSnapshots.length - 1],
+    mostRecentSnapshot: snapshots[snapshots.length - 1],
   });
 
-  return awsResponse.DBSnapshots[awsResponse.DBSnapshots.length - 1].DBSnapshotIdentifier!;
+  return snapshots[snapshots.length - 1]?.DBSnapshotIdentifier;
+}
+
+/**
+ * @description Create DB snapshot for an RDS database. Only returns when the
+ * snapshot is available.
+ */
+export async function createDBSnapshot(
+  rds: RDS,
+  snapshotIdentifier: string,
+  dbInstanceIdentifier: string,
+): Promise<string> {
+  const params = {
+    DBInstanceIdentifier: dbInstanceIdentifier,
+    DBSnapshotIdentifier: snapshotIdentifier,
+  };
+
+  try {
+    await rds.createDBSnapshot(params).promise();
+    // Polling function to check snapshot status. Only return when the snapshot is available.
+    const waitForSnapshot = async () => {
+      // eslint-disable-next-line no-constant-condition
+      while (true) {
```

Comment:
nit: Add some max wait time where the task breaks out of this loop, and some logs indicating how long the task is waiting for the snapshot at some X interval to help with debugging if this loops somehow never terminates.

Reply:
changed to use waiter in AWS SDK, that sets a max time limit (retries * delay)